lib/rules/disallow-variable-names.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Disallows a specified set of variable names.
+ *
+ * Type: `Array`
+ *
+ * Values: Array of strings, which should be disallowed as variable names
+ *
+ * #### Example
+ *
+ * ```js
+ * "disallowVariableNames": ['temp', 'foo']
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * var good = 1;
+ * object[fine] = 2;
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * var temp = 1;
+ * object[foo] = 2;
+ * ```
+ */
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+
+    configure: function(disallowVariableNames) {
+        assert(
+            Array.isArray(disallowVariableNames),
+            'disallowVariableNames option requires an array'
+        );
+
+        this._disallowVariableNames = disallowVariableNames;
+    },
+
+    getOptionName: function() {
+        return 'disallowVariableNames';
+    },
+
+    check: function(file, errors) {
+        var disallowVariableNames = this._disallowVariableNames;
+        file.iterateNodesByType('Identifier', function(node) {
+            if (disallowVariableNames.indexOf(node.name) !== -1) {
```

Comment:
@mikesherov any opinion on the uppercase/lowercase issue above?

@alawatthe I was iffy on mentioning the object notation for the same fear of "the root of all evil," but:

1. This is one of the more obvious improvements that doesn't obfuscate the code (it actually clarifies the condition).
2. We have other rules that do this: https://github.com/jscs-dev/node-jscs/blob/master/lib/rules/disallow-dangling-underscores.js#L84-L87, https://github.com/jscs-dev/node-jscs/blob/master/lib/rules/disallow-implicit-type-conversion.js#L41-L44, https://github.com/jscs-dev/node-jscs/blob/master/lib/rules/disallow-keywords.js#L31-L34, and a bunch more.

We'll wait for other opinions on casing to decide on the course of action.